lightning/src/util/config.rs

```diff
@@ -21,7 +21,14 @@ pub struct ChannelHandshakeConfig {
 	/// Applied only for inbound channels (see ChannelHandshakeLimits::max_minimum_depth for the
 	/// equivalent limit applied to outbound channels).
 	///
+	/// A lower-bound of 1 is applied, requiring all channels to have a confirmed commitment
+	/// transaction before operation. If you wish to accept channels with zero confirmations, see
+	/// [`UserConfig::manually_accept_inbound_channels`] and
+	/// [`ChannelManager::accept_inbound_channel`].
```

Comment:
Should we link to `accept_inbound_channel_from_trusted_peer_0conf`? I think the API  might be cleaner if we aggregate the two methods .. Also so weird to me that you can't initiate 0-conf channels, only accept them 🤔

Reply:
This is "applied only for inbound channels" :)

Reply:
I guess not entirely applicable to the comment, but perhaps explaining why a config parameter would go in `ChannelHandshakeConfig` vs `ChannelHandshakeLimits`. The comments on the fields in the `UserConfig` make it clear, but `ChannelHandshakeConfig`'s docs are super vague. Feel free to disregard as it is unrelated to this PR.